File: tcm/model/message.go
package model

type Gr25Message struct {
	Timestamp  int    `json:"timestamp"`
	VehicleID  string `json:"vehicle_id"`
	Topic      string `json:"topic"`
	Data       []byte `json:"data" gorm:"type:bytea"` // PostgreSQL uses BYTEA for binary data
	Synced     int    `json:"synced"`
	SourceNode string `json:"source_node"`
	TargetNode string `json:"target_node"`
}

func (Gr25Message) TableName() string {
	return "gr25_message"
}

File: shelter/model/message.py
import polars as pl


GR26_MESSAGE_SCHEMA = pl.Schema(
    {
        "timestamp": pl.Int64,
        "vehicle_id": pl.Utf8,
        "topic": pl.Utf8,
        "data": pl.Binary,
        "source_node": pl.Utf8,
        "target_node": pl.Utf8,
    }
)
